ArduCopter/Copter.h

```diff
@@ -626,7 +626,31 @@ class Copter : public AP_Vehicle {
     static_assert(_failsafe_priorities[ARRAY_SIZE(_failsafe_priorities) - 1] == -1,
                   "_failsafe_priorities is missing the sentinel");
 
+    // previous command types
+    enum class PrevCmd : uint8_t {
+        DO_USER_TAKEOFF_START = 0,
+        SET_DESTINATION_VECTOR = 1,
+        SET_DESTINATION_LOCATION = 2,
+        SET_DESTINATION_POSVEL = 3,
+        SET_DESTINATION_POSVELACCEL = 4,
+    };
 
+    // previous commands from GCS
```

Comment:
This state should all be contained within the ModeGuided class.

Same with prevCmd enum